### DISCUSSION TITLE
Rendering ModelViewer in AndroidView

### DISCUSSION THREAD

**Discussion by paulosensorydata** [google/filament#4595] [Q&A]
Hey everyone. I am trying to update an AndroidView based on my CustomViewModel changes - I am changing a parameter there and I want to change the geometry/renderable I previously rendered in the ModelViewer scene:

```
@Composable
fun FilamentViewer(viewModel: CustomViewModel) {
    var modelViewer by remember { mutableStateOf<ModelViewer?>(null) }

    launchInComposition {
        while (true) {
            withFrameNanos { frameTimeNanos ->
                modelViewer?.render(frameTimeNanos)
            }
        }
    }

    AndroidView({ context ->
        LayoutInflater.from(context).inflate(
            R.layout.filament_host, FrameLayout(context), false
        ).apply {
            modelViewer = renderFilamentView(this, context, viewModel.values)
        }
    })
}
```

When `viewModel.values` are updated, I need to change at least the scene renderables (maybe remove the already-existing one) and add the new one? However I was not successful yet - already tried to use the update/onUpdate argument of the `AndroidView`. Can this be achieved?

**Comment by romainguy** [comment]
You can see an example here, using `onCommit()`: https://github.com/romainguy/sample-materials-shop/blob/main/app/src/main/kotlin/com/curiouscreature/compose/sample/shop/MainActivity.kt#L324 This was however for an older version of Jetpack Compose. Now you'd want to use `SideEffect` instead. See https://developer.android.com/jetpack/compose/side-effects#sideeffect-publish

**Reply by paulosensorydata** [reply]
`SideEffect/onCommit()` only runs once, right? In this case, my `customViewModel` is being updated every 2 seconds

**Reply by romainguy** [reply]
No, `SideEffect` will [run on every recomposition](https://developer.android.com/jetpack/compose/side-effects#sideeffect-publish).
